### PR TITLE
new internal register event API

### PR DESCRIPTION
**Pull request by MariusDieckmann**
Internal API for registering events into the system

### CODE REVIEW

File: aruna/api/internal/v1/notification.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    rpc RegisterEvent(RegisterEventRequest) returns (RegisterEventResponse) {}
+}
+
+message RegisterEventRequest{
+    EventResource event_resource = 1;
+    string resource_id = 2;
+    EventType event_type = 3;
+}
+
+enum EventResource {
```
Comment (St4NNi) [inline aruna/api/internal/v1/notification.proto:18]: Maybe this should also include `Projects`

File: aruna/api/internal/v1/notification.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    EVENT_RESOURCE_OBJECT = 3;
+    EVENT_RESOURCE_ALL = 5;
+}
+  
+enum EventType {
+    EVENT_TYPE_UNSPECIFIED = 0;
+    EVENT_TYPE_CREATED = 1;
+    EVENT_TYPE_AVAILABLE = 2;
+    EVENT_TYPE_UPDATED = 3;
+    EVENT_TYPE_METADATA_UPDATED = 4;
```
Comment (St4NNi) [inline aruna/api/internal/v1/notification.proto:31]: What should "Metadata Updated" be used for ?

File: aruna/api/notification/services/v1/notification_service_models.proto [hunk 3]
```diff
@@ -59,22 +53,23 @@ message NotificationStreamResponse {
...
-    UPDATE_TYPE_UNSPECIFIED = 0;
-    UPDATE_TYPE_CREATED = 1;
-    UPDATE_TYPE_AVAILABLE = 2;
-    UPDATE_TYPE_UPDATED = 3;
-    UPDATE_TYPE_METADATA_UPDATED = 4;
-    UPDATE_TYPE_DELETED = 5;
-  }
+  EventType updated_type = 3;
+}
+enum EventResource {
```
Comment (St4NNi) [inline aruna/api/notification/services/v1/notification_service_models.proto:60]: We should not include redundant definitions of EventResource and EventType

**Review by St4NNi — CHANGES_REQUESTED** [review]
Good ideas but some improvements should be made before merging.

- Remove redundant definitions of `EventResource` and `EventType`
- Add `Project` to `EventResource` ?
- "authorize" method to authorize event access ? / Internal API

File: aruna/api/internal/v1/notification.proto (hunk 1 above)
Comment (St4NNi) [reply aruna/api/internal/v1/notification.proto]: Solved by [47a9bf7](https://github.com/ArunaStorage/ArunaAPI/pull/39/commits/47a9bf74d43af884769f58a39f9208352c11526d) and [72facaf](https://github.com/ArunaStorage/ArunaAPI/pull/39/commits/72facaf08a2ec5a39366389dad1fa1ed4cd0c120)

**Review by St4NNi — APPROVED** [review]

**Review by MariusDieckmann — COMMENTED** [review]
Minor changes required, mostly documentation

File: aruna/api/notification/services/v1/notification_service.proto [hunk 4]
```diff
@@ -5,15 +5,98 @@ option go_package = "github.com/ArunaStorage/go-api/aruna/api/storage/services/v
...
 
+  // DeleteEventStreamingGroup
+  //
+  // Deletes a existing EventStreamingGroup by ID
+  rpc DeleteEventStreamingGroup(DeleteEventStreamingGroupRequest)
+      returns (DeleteEventStreamingGroupResponse) {}
+
+  // NotificationStreamGroup
+  //
+  // Requests a NotificationStreamGroup
```
[the remark this answers is not in the document]
Comment (St4NNi) [reply aruna/api/notification/services/v1/notification_service.proto]: Solved by [25006c8](https://github.com/ArunaStorage/ArunaAPI/pull/39/commits/25006c856dd0f534402cfb20238aa4ea38f7a3a7)

File: aruna/api/internal/v1/authorize.proto [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+message Authorization {
+    // Can be empty for now
+    string secretkey = 1;
+    // Should include the APItoken
+    string accesskey = 2;
+    // If the secret is an APIToken (or S3-style access / secretkey)
+    bool is_token = 3;
+}
+
+message AuthorizeRequest{
```
[the remark this answers is not in the document]
Comment (St4NNi) [reply aruna/api/internal/v1/authorize.proto:23]: Solved by [25006c8](https://github.com/ArunaStorage/ArunaAPI/pull/39/commits/25006c856dd0f534402cfb20238aa4ea38f7a3a7)

File: aruna/api/internal/v1/authorize.proto [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+import "google/api/visibility.proto";
+import "aruna/api/storage/models/v1/models.proto";
+
+service InternalAuthorizeService {
+    option (google.api.api_visibility).restriction = "INTERNAL";
+    rpc Authorize(AuthorizeRequest) returns (AuthorizeResponse) {}
+}
+
+message Authorization {
```
[the remark this answers is not in the document]
Comment (St4NNi) [reply aruna/api/internal/v1/authorize.proto:14]: Solved by [25006c8](https://github.com/ArunaStorage/ArunaAPI/pull/39/commits/25006c856dd0f534402cfb20238aa4ea38f7a3a7)

**Review by MariusDieckmann — COMMENTED** [review]
Approved

**Pull request merged**